client/js/utils/utils.js

```diff
@@ -0,0 +1,32 @@
+var _ = {};
+
+_.each = function(collection, callback){
+  if(collection instanceof Array){
+    for(var i = 0; i < collection.length; i++){
+      callback(collection[i], i, collection);
+    }
+  }else{
+    for(var key in collection){
+      callback(collection[key], key, collection);
+    }
+  }
+};
+
+_.map = function(collection, callback){
+  var result = [];
+  _.each(collection, function(item, key, collection){
+    result.push(callback(item, key, collection));
+  });
+  return result;
+};
+
+_.reduce = function(collection, callback, accumulator){
+  var start = accumulator === undefined ? 1 : 0;
+  accumulator = accumulator === undefined ? collection[0] : accumulator;
+  _.each(collection, function(item, key, collection){
+    accumulator = start - 1 === key ? accumulator : callback(accumulator, item);
```

Comment:
just for reference - the initialization of accumulators here will not work for objects, since objects don't necessarily have numerical keys. also, even if the POJO had numerical keys, they will be strings: start - 1 will evaluate to number, and the key will be of type string, so that will never return true.